addons/sale_pdf_quote_builder/models/ir_actions_report.py

```diff
@@ -101,50 +131,78 @@ def _get_form_fields_mapping(self, order, doc_line_id_mapping=None):
+        """ Format a field value from the extracted PDF string.
 
-        Override this method to add new fields to the mapping.
+        :param string form_field_name: field path
+        :param recordset order: sale.order record from which to take the value, following the path
+        :param dict translated_env: self.env, with the correct translation context to format amount,
+                                    date and datetime
+        :param str tz: the timezone used for rendering datetime
+        :param dict param_field_map: the map stored in the config parameter to assign the expected
+                                     paths to the form fields. A form field not in that map will be
+                                     ignored.
+        :rtype: string
+        :return: formatted field value
 
-        :param recordset line: sale.order.line record
-        :rtype: dict
-        :return: mapping of prefixed fields name to Odoo fields data
-
-        Note: line.ensure_one()
+        Note: order.ensure_one()
         """
-        line.ensure_one()
-        env = self.with_context(use_babel=True).env
-        return {
-            f'{line.id}_description': line.name,
-            f'{line.id}_quantity': line.product_uom_qty,
-            f'{line.id}_uom': line.product_uom.name,
-            f'{line.id}_price_unit': format_amount(env, line.price_unit, line.currency_id),
-            f'{line.id}_discount': line.discount,
-            f'{line.id}_product_sale_price': format_amount(
-                env, line.product_id.lst_price, line.product_id.currency_id
-            ),
-            f'{line.id}_taxes': ', '.join(tax.name for tax in line.tax_id),
-            f'{line.id}_tax_excl_price': format_amount(env, line.price_subtotal, line.currency_id),
-            f'{line.id}_tax_incl_price': format_amount(env, line.price_total, line.currency_id),
-        }
+        def _get_formatted_value(field_type_, value_):
+            if field_type_ == 'boolean':
+                formatted_value = _("Yes") if value_ else _("No")
+            elif field_type_ == 'monetary':
+                currency_id = records[field.get_currency_field(records)]
+                formatted_value = format_amount(
+                    translated_env, value_, currency_id or order.currency_id
+                )
+            elif field_type_ == 'date':
+                formatted_value = format_date(translated_env, value_, lang_code=lang)
+            elif field_type_ == 'datetime':
+                formatted_value = format_datetime(translated_env, value_, tz=tz)
+            elif field_type_ == 'selection' and value_:
+                formatted_value = dict(field._description_selection(translated_env))[value_]
+            elif field_type_ in {'one2many', 'many2one', 'many2many'}:
+                formatted_value = ', '.join([v.display_name for v in value_])
+            else:
+                formatted_value = value_
```

Comment:
Well, my remark on `field_type` was mostly because you use parameters before their definition, which is confusing at first glance.
Maybe we could move the method some lines below, so that the 'constant' variables it uses are defined first.

Reply:
Hm, having inner methods in the middle of a method makes no sense to me. Either on top (as it's done here), or I extract it entirely (which isn't needed).

Reply:
Well, conceptually, it's like defining a variable only when it has to be used and not before the checks and asserts that happen before it's used anyway.

Reply:
Conceptually, inner method do have access to variables in the outer method =]